src/master-playlist-controller.js

```diff
@@ -1557,4 +1582,32 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     AdCueTags.updateAdCues(media, this.cueTagsTrack_, offset);
   }
+
+  /**
+   * Calculates the desired forward buffer length based on current time
+   *
+   * @return {Number} Desired forward buffer length in seconds
+   */
+  goalBufferLength() {
```

Comment:
There is some duplication of this function in the test suite.  You could:

1.Take currentTime in as a variable into the function and use that for testing OR
2. Take currentTime in and set it to a default value, then change it/set it in the test.

This would also let you write a couple of tiny unit tests to check the internal logic of this function if you wanted to.

You could probably do the same for bufferLowWaterLine if you wanted to test it similarly.